src/main/scala/au/com/agiledigital/toolform/command/generate/docker/GenerateDockerComposeV3Command.scala

```diff
@@ -121,18 +121,24 @@ object GenerateDockerComposeV3Command extends YamlWriter {
           }
     } yield ()
 
-  def writeResource(resource: Resource): Result[Unit] =
+  def writeResource(resource: Resource): Result[Unit] = {
+    val imageName = resource.image match {
+      case Some(name) => name
+      case None       => throw new Exception(s"""Image name does not exist for [${resource.id}]""")
```

Comment:
We try to avoid using exceptions in Scala as a style rule. They are more of a concept inherited from Java and they break the strong type system of Scala. If you use an `Either` type instead, the fact the the method can throw an error is encoded by the type system and the programmer has to handle it somewhere or the code won't compile.

However, it looks like it might be complicated to make this method return an `Either` type because it is using a state monad (the `Result` type is defined by `type Result[A] = State[WriterContext, A]`). The state monad lets us chain together all these write functions in for comprehensions and pass a context object around which maintains the writer state in a clean way. It seems like it would be way too much trouble to introduce an either into the middle of all that since we would have introduce another level of abstraction. 

It would be much easier if we could push down that validation into a part of the code that is already set up for handling errors. For example `runGenerateDockerComposeV3` already has `Either[NonEmptyList[ToolFormError], String]` as its return type. If you could return the validation error at that stage, before it gets into the state monad section (the block executed by `run`) than it would be handled by the existing error handling infrastructure.

I know that there are exceptions thrown elsewhere in the code base but we are trying to avoid them going forward.